### PR TITLE
Add matrix for msm8998

### PR DESCRIPTION
**Pull request by phhusson**
This has been done by looking at log history of msm8998.dtsi of when nodes are added

Things I'm not sure:
- status-connectivity-ipa: N /// I'm under the impression that msm8996 should have it, but it's N/A, so I'm not sure what it means
- smp: msm8996 has the additional comments which might also apply
- Is spm the same thing as spmi?
- What's sdsp?
- Does fastrpc require explicit declaration in dts? I'm not sure what fastrpc is
- Does smmu apply to /any/ smmu? All? Maybe they got all merged together i don't know
- I merged usb host mode and slave mode since it's a dwc which should be generic, but maybe that's wrong

### CODE REVIEW

**Comment by lumag** [comment]
Thank you for your patches. Let's resolve your questions first.

> status-connectivity-ipa: N /// I'm under the impression that msm8996 should have it, but it's N/A, so I'm not sure what it means

we should fix msm8996, indeed.

> smp: msm8996 has the additional comments which might also apply

I don't think so, msm8998 didn't have the story of the CBF clock.

> Is spm the same thing as spmi?

No, SPM is the SPM / SAW2, a part of the power management. It is handled in a different way on MSM8998, so just ignore it.

> What's sdsp?

Sensors DSP aka SLPI.

> Does fastrpc require explicit declaration in dts? I'm not sure what fastrpc is

fastrpc is a special interface between userspace and the DSP. It is enabled by the `qcom,fastrpc-compute-cb` compatible.

> Does smmu apply to /any/ smmu? All? Maybe they got all merged together i don't know

Usually yes. There are some platforms where one of SMMU works, other do not. This can be reflected via WIP or status-smmu-comment.

> I merged usb host mode and slave mode since it's a dwc which should be generic, but maybe that's wrong

If it really works for you, please. Newer platforms require special handling for the host mode on the Type-C host, that was the reason for the split.

**Comment by phhusson** [comment]
Thanks for your feedback. I think it should be all good now.

> If it really works for you, please. Newer platforms require special handling for the host mode on the Type-C host, that was the reason for the split.

Sadly my device (Freebox Devialet) has host-only, so I can't really test slave. However I seem to remember trying slave on fxtec Pro1 and succeeding, but that was a long time ago.

File: _soc/msm8998.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+status-storage-nand: N/A
+status-storage-sata: N/A
+status-storage-sdcc: 4.20
+status-storage-ufs: 5.1
+status-uart: 4.19
+status-usb: 5.0
+status-usb-hostmode: 5.0
+status-usb-periphmode: 5.0
+status-video-venus: N
+pmic: pm8998
```
Comment (lumag) [inline _soc/msm8998.md]:
```suggestion
pmic: pm8998, pmi8998, pm8005
```

**Comment by lumag** [comment]
@phhusson thanks, this looks good now.

**Pull request merged**